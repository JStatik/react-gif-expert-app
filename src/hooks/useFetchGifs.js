import { useState, useEffect } from 'react';
import getGifs from '../helpers/getGifs';

const useFetchGifs = ( categoria ) => {
    const [ informacion, setInformacion ] = useState( { imagenes: [], cargando: true } );

    useEffect( () => {
        getGifs( categoria ).then( ( gifs ) => setInformacion( { imagenes: gifs, cargando: false } ) );        
    }, [ categoria ] );

    return informacion;
};

export default useFetchGifs;
