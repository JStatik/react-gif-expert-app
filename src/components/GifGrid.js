import React from 'react';
import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';
import useFetchGifs from '../hooks/useFetchGifs';

const GifGrid = ( { categoria } ) => {
    const { imagenes, cargando } = useFetchGifs( categoria );

    return (
        <>  
            <h3 className="centrado animate__animated animate__jackInTheBox">{ categoria }</h3>

            <p className="centrado">{  cargando && 'Cargando...' }</p>

            <div className="card-grid">            
                {
                    imagenes.map( ( elem ) => {
                        return <GifGridItem key={ elem.id } { ...elem } />
                    } )
                }
            </div>
        </>
    )
};

GifGrid.propTypes = {
    categoria: PropTypes.string.isRequired
}

export default GifGrid;
