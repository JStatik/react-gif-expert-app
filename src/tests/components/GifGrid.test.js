import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock( '../../hooks/useFetchGifs' );

describe( 'Pruebas en el componente GifGrid', () => {    
    test( 'Debe mostrar el componente sin imagen alguna y con el mensaje "Cargando..."', () => {
        useFetchGifs.mockReturnValue( {
            imagenes: [],
            cargando: true
        } );

        const wrapper = shallow( <GifGrid categoria="Naruto Shippuden" /> );

        expect( wrapper ).toMatchSnapshot();
    } );

    test( 'Debe mostrar una imagen en el componente correctamente y ocultar el mensaje "Cargando..."; adicionalmente verificamos que el parrafo este vacio, que el Componente GifGridItem exista y que la cantidad de componentes GifGridItem sea los mismos que los gifs obtenidos por useFetchGifs', () => {
        const gifs = [ {
            id: 'abc123',
            url: 'https://localhost/naruto+shippuden/gif',
            title: 'Naruto'
        } ];

        useFetchGifs.mockReturnValue( {
            imagenes: gifs,
            cargando: false
        } );

        const wrapper = shallow( <GifGrid categoria="Naruto Shippuden" /> );

        expect( wrapper ).toMatchSnapshot();

        expect( '' ).toBe( wrapper.find( 'p' ).text() );
        expect( true ).toBe( wrapper.find( 'GifGridItem' ).exists() );
        expect( gifs.length ).toBe( wrapper.find( 'GifGridItem' ).length );
    } );
} );
