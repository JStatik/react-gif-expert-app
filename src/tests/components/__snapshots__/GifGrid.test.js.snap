// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe mostrar el componente sin imagen alguna y con el mensaje "Cargando..." 1`] = `
<Fragment>
  <h3
    className="centrado animate__animated animate__jackInTheBox"
  >
    Naruto Shippuden
  </h3>
  <p
    className="centrado"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente GifGrid Debe mostrar una imagen en el componente correctamente y ocultar el mensaje "Cargando..."; adicionalmente verificamos que el parrafo este vacio, que el Componente GifGridItem exista y que la cantidad de componentes GifGridItem sea los mismos que los gifs obtenidos por useFetchGifs 1`] = `
<Fragment>
  <h3
    className="centrado animate__animated animate__jackInTheBox"
  >
    Naruto Shippuden
  </h3>
  <p
    className="centrado"
  />
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc123"
      key="abc123"
      title="Naruto"
      url="https://localhost/naruto+shippuden/gif"
    />
  </div>
</Fragment>
`;
